scsSDKTelemetry.cc

```diff
@@ -43,37 +49,65 @@ napi_value GetBuffer(napi_env env, napi_callback_info info) {
     return nullptr;
   }
 
+#if defined(_WIN32)
   // Open the memory-mapped file
-  hMapFileSCSTelemetry = OpenFileMapping(FILE_MAP_ALL_ACCESS, FALSE, mmf_name);
+  HANDLE hMapFileSCSTelemetry = OpenFileMapping(FILE_MAP_READ, FALSE, mmf_name);
   free(mmf_name);
   if (!hMapFileSCSTelemetry) {
     napi_throw_error(env, NULL, "Failed trying to open memory-mapped file");
     return nullptr;
   }
 
-  // Map the file into the process's address space
-  pBuf = MapViewOfFile(hMapFileSCSTelemetry, FILE_MAP_ALL_ACCESS, 0, 0, mmf_size);
-  if (!pBuf) {
+  mapped = MapViewOfFile(hMapFileSCSTelemetry, FILE_MAP_READ, 0, 0, mmf_size);
```

Comment:
The term `mapped` seems a bit confusing here, but so was the original `pBuf`, I suggest naming this `mappedFileView` to reflect what it actually is
```suggestion
  mappedFileView= MapViewOfFile(hMapFileSCSTelemetry, FILE_MAP_READ, 0, 0, mmf_size);
```